### ISSUE TITLE
ZIO 2.0: Composable ZIO#validate

### ISSUE THREAD

**Issue by khajavi** [zio/zio#6421]
The current implementation of `ZIO#validate` is nested and the unit values contribute to the output:

```scala
val r1: ZIO[Any, Throwable, ((Int, Int), Int)]    = ZIO(1) validate ZIO(2) validate ZIO(3)
val r2: ZIO[Any, Throwable, ((Unit, Unit), Unit)] = ZIO(1).unit validate ZIO(2).unit validate ZIO(3).unit
```

I'm thinking it would be better to have a consistent API like the [zips](https://github.com/zio/zio/issues/5267).

**Comment by adamgfraser** [comment]
Thanks for reporting! Will take a look.

**Comment by adamgfraser** [comment]
Closed by #6423.
